### PR TITLE
try to use current site for relative_urls

### PR DESCRIPTION
**Pull request by georgewhewell**
If current site and another both have the same root path, we should prefer to create a relative link to current site

### CODE REVIEW

File: wagtail/wagtailcore/models.py [hunk 1]
```diff
@@ -593,6 +593,9 @@ def relative_url(self, current_site):
         a local URL if the site matches, or a fully qualified one otherwise.
         Return None if the page is not routable.
         """
+        if self.url_path.startswith(current_site.root_page.url_path):
+            return reverse('wagtail_serve', args=(self.url_path[len(current_site.root_page.url_path):],))
+
```
Comment (kaedroho) [inline wagtail/wagtailcore/models.py]: Could you remove the indentation from line 598?

**Comment by kaedroho** [comment]
Could you update this test? https://github.com/torchbox/wagtail/blob/f2e4832ca957b9d0d0a558c8fb4018c56bd34197/wagtail/wagtailcore/tests/test_page_model.py#L140-L158

**Comment by gasman** [comment]
Testing `self.url_path.startswith` is not sufficient, because it's possible for one site root to exist underneath another, as in the currently-failing unit test. In this situation, the page is considered to exist in whichever site is deepest in the tree.

This does mean that behaviour is currently undefined when two sites have the same root. I wouldn't be opposed to changing this so that it has some well-defined behaviour, but:

* it would need some additional unit tests specifically addressing that
* at some point, I'd like to refactor the whole `url` / `full_url` / `relative_url` thing so that they work off a common base method that returns a (site, path) tuple. It could be tricky to make that work with the behaviour proposed here, because we're side-stepping the decision about which site the page really belongs to. (That's not a showstopper, though: the base method just needs to choose one site or the other, and the rule for `relative_url` will then be "output the full URL unless the page's site has a root_path that matches `current_site`'s root_path".)

See also: #1283

**Comment by gasman** [comment]
Having added some further thoughts at https://github.com/torchbox/wagtail/issues/1283#issuecomment-100470750, I think it might be worth taking a step back and thinking about what the use-case is for having multiple sites pointing to the same root, as that's going to dictate what the desired behaviour actually is.

If it's about having different hostnames for different environments (e.g. live vs staging), then the idea proposed on that ticket might work. Or if it's about allowing alternative URLs to the same place (such as www vs non-www), then maybe it should really be a single Site record consisting of a main hostname and zero or more aliases...

**Comment by gasman** [comment]
In the absence of any clear agreement on what the use-case is for having multiple sites at the same root, I don't think this can really be progressed - closing.

**Pull request closed without merging**